radio/bluetooth/bluetooth.go

```diff
@@ -0,0 +1,88 @@
+package bluetooth
+
+import (
+	"log"
+	"time"
+
+	"github.com/paypal/gatt"
+	"github.com/paypal/gatt/examples/option"
+)
+
+/*
+Uses the gatt package
+https://github.com/paypal/gatt
+*/
+
+var radio gatt.Device
+var bluetoothChannel = make(chan gatt.Peripheral)
+
+func Scan(name string, timeout time.Duration) ([]string, error) {
+	log.Printf("Scanning for bluetooth devices named %s\r\n", name)
+	initialise()
+	radio.Handle(gatt.PeripheralDiscovered(onPeriphDiscovered))
+	radio.Init(onStateChanged)
+
+	devices := make([]string, 0, 10)
+
+	for {
+		select {
+		case <-time.After(timeout * time.Second):
+			radio.StopScanning()
+			return devices, nil
+		case onlineDevice := <-bluetoothChannel:
+			if onlineDevice.Name() == name {
+				devices = append(devices, onlineDevice.ID())
+			}
+		}
+	}
+}
+
+func Online(id string, timeout time.Duration) (bool, error) {
+	log.Printf("Checking if bluetooth device %s is online\r\n", id)
+	initialise()
```

Comment:
Again with the arbitrary `initialise()`, either use `init()` or have a specific function to do this run somewhere else. Having an unrelated function perform initialisation can result in hard to understand bugs 'I was just checking for online devices, what's wrong with online device checking?!'